roles/valheim/defaults/main.yml

```diff
@@ -0,0 +1,132 @@
+##########################################################################
+# Title:         Sandbox: valheim | Default Variables                    #
+# Author(s):     scarabyte                                               #
+# URL:           https://github.com/saltyorg/Sandbox                     #
+# --                                                                     #
+##########################################################################
+#                   GNU General Public License v3.0                      #
+##########################################################################
+---
+################################
+# Basics
+################################
+
+valheim_name: valheim
+valheim_version: "LATEST"
+
+################################
+# Paths
+################################
+
+valheim_paths_folder: "{{ valheim_name }}"
+valheim_paths_location: "{{ server_appdata_path }}/{{ valheim_paths_folder }}"
+valheim_paths_folders_list:
+  - "{{ valheim_paths_location }}"
+
+################################
+# Web
+################################
+
+valheim_web_subdomain: "{{ valheim_name }}"
+valheim_web_domain: "{{ user.domain }}"
+
+################################
+# DNS
+################################
+
+valheim_dns_record: "{{ lookup('vars', valheim_name + '_web_subdomain', default=valheim_web_subdomain) }}"
+valheim_dns_zone: "{{ lookup('vars', valheim_name + '_web_domain', default=valheim_web_domain) }}"
+valheim_dns_proxy: false
+
+################################
+# Docker
+################################
+
+# Container
+valheim_docker_container: "{{ valheim_name }}"
+
+# Image
+valheim_docker_image_pull: true
+valheim_docker_image_tag: "latest"
+valheim_docker_image: "mbround18/valheim:{{ valheim_docker_image_tag }}"
+
+# Ports
+valheim_docker_ports_defaults:
+  - "2456:2456/udp"
+  - "2457:2457/udp"
+  - "2458:2458/udp"
+valheim_docker_ports: "{{ valheim_docker_ports_defaults }}"
+
+# Envs
+valheim_docker_envs_default:
+  TZ: "{{ tz }}"
+  UID: "{{ uid }}"
+  GID: "{{ gid }}"
+  PUBLIC: "0"
+  PASSWORD: "{{ lookup('community.general.random_string') }}"
```

Comment:
This should be a separate variable so users can override it if they want. Both PUBLIC and PASSWORD, imho.

Reply:
Could you link an example on how to do this best? I had a variable for the password but when using it in there server name and password env parts they were both different.

Reply:
The tandoor role requires a secret key that is set in the sandbox settings.yml file. That could be an example for the password. Agree with salty, the public bit could be a simple variable.